src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -25,23 +28,25 @@ public class ShoppingCartController {
     public ShoppingCartController(ShoppingCartService shoppingCartService,
                                   UserService userService,
                                   MovieSessionService movieSessionService,
-            ResponseDtoMapper<ShoppingCartResponseDto, ShoppingCart>
-                                      shoppingCartResponseDtoMapper) {
+                                  ResponseDtoMapper<ShoppingCartResponseDto, ShoppingCart>
+                                          shoppingCartResponseDtoMapper) {
         this.shoppingCartService = shoppingCartService;
         this.userService = userService;
         this.movieSessionService = movieSessionService;
         this.shoppingCartResponseDtoMapper = shoppingCartResponseDtoMapper;
     }
 
     @PutMapping("/movie-sessions")
-    public void addToCart(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+    public void addToCart(@RequestBody @Valid Authentication authentication,
+                          @RequestParam @Valid Long movieSessionId) {
```

Comment:
There is no need to validate the authentication object as we do it at the registration stage. I mean @Valid is redundant and so is @RequestBody